nix/default.nix

```diff
@@ -274,18 +317,48 @@ let
           stdio
           uri
           uutf
-          zarith # cmdliner easy-format js_of_ocaml ocp-ocamlres tls
+          zarith
+          cmdliner
+          # easy-format js_of_ocaml ocp-ocamlres tls
           # alcotest dum pprint
+          ocp-ocamlres
+          pprint
           ocamlgraph
           findlib
           genspio
         ] ++ [ libusb1 libusb1.out (gmp.override { withStatic = true; }) ];
         doCheck = false;
-        buildPhase = "dune build src/bin_client/tezos-client.install";
+        buildPhase = ''
+          # tezos-node build requires ocp-ocamlres binary in PATH
+          PATH=$PATH:${ocp-ocamlres}/lib/ocaml/4.07.1/bin
+          dune build src/bin_client/tezos-client.install
+          dune build src/bin_node/tezos-node.install
+          dune build src/proto_${branchInfo.protoName}/bin_baker/tezos-baker-${branchInfo.binarySuffix}.install
+          dune build src/proto_${branchInfo.protoName}/bin_accuser/tezos-accuser-${branchInfo.binarySuffix}.install
+          dune build src/proto_${branchInfo.protoName}/bin_endorser/tezos-endorser-${branchInfo.binarySuffix}.install
+          dune build src/bin_signer/tezos-signer.install
+          dune build src/lib_protocol_compiler/tezos-protocol-compiler.install
+        '';
         installPhase = ''
           mkdir -p $out/bin
+          # tezos-client and tezos-admin
           cp _build/default/src/bin_client/main_client.exe $out/bin/tezos-client
-          cp _build/default/src/bin_client/main_admin.exe $out/bin/tezos-admin
+          cp _build/default/src/bin_client/main_admin.exe $out/bin/tezos-admin-client
```

Comment:
maybe just loop over these produced exes? are they always called main_?

Reply:
> are they always called main_?
Nope, `tezos-node` has a simple `main.exe` executable.
So it seems that loop over `main*.exe` is something non-trivial since we cannot always get `tezos-smth` name by the `.exe` name